docs/userguide/networking/dockernetworks.md

```diff
@@ -0,0 +1,368 @@
+172.17.0.3      container2
+172.17.0.3      container2.bridge
+/ #
+
+```
+
+The default `docker0` bridge network supports the use of port mapping and  `docker run --link` to allow communications between containers in the `docker0` network. These techniques are cumbersome to set up and prone to error. While they are still available to you as techniques, it is better to avoid them and define your own networks instead.
+
+## User-defined networks
+
+Users can create their own user-defined networks that better isolate containers.  Docker provides some default **network drivers** for use defining these networks.  You can create a new **bridge network** or **overlay network**. You can also create a **network plugin** or **remote network**  written to your own specifications.
+
+The next few sections describe each of these drivers in greater detail.
+
+### A bridge network
+
+The easiest user-defined network to create is a `bridge` network. This network is similar to the historical, default `docker0` network. There are some added features and some old features that aren't available.
+
+```
+$ docker network create -d bridge isolated_nw
+8b05faa32aeb43215f67678084a9c51afbdffe64cd91e3f5bb8267475f8bf1a7
+
+$ docker network inspect isolated_nw
+{
+    "name": "isolated_nw",
+    "id": "8b05faa32aeb43215f67678084a9c51afbdffe64cd91e3f5bb8267475f8bf1a7",
+    "driver": "bridge",
+    "containers": {}
+}
+
+$ docker network ls
+NETWORK ID          NAME                DRIVER
+9f904ee27bf5        none                null
+cf03ee007fb4        host                host
+7fca4eb8c647        bridge              bridge
+8b05faa32aeb        isolated_nw         bridge
+
+```
+
+After you create the network, you can launch containers on it using  the `docker run --net=<NETWORK>` option.
+
+```
+$ docker run --net=isolated_nw -itd --name=container3 busybox
+777344ef4943d34827a3504a802bf15db69327d7abe4af28a05084ca7406f843
+
+$ docker network inspect isolated_nw
+{
+    "name": "isolated_nw",
+    "id": "8b05faa32aeb43215f67678084a9c51afbdffe64cd91e3f5bb8267475f8bf1a7",
+    "driver": "bridge",
+    "containers": {
+        "777344ef4943d34827a3504a802bf15db69327d7abe4af28a05084ca7406f843": {
+            "endpoint": "c7f22f8da07fb8ecc687d08377cfcdb80b4dd8624c2a8208b1a4268985e38683",
+            "mac_address": "02:42:ac:14:00:01",
+            "ipv4_address": "172.20.0.1/16",
+            "ipv6_address": ""
+        }
+    }
+}
+```
+
+The containers you launch into this network must reside on the same Docker host.  Each container in the network can immediately communicate with other containers in the network. Though, the network itself isolates the containers from external networks.
+
+![](images/bridge_network.png)
+
+Within of a user-defined bridge network, linking is not supported. You can expose and publish container ports on containers in this network. This is useful if you want make a portion of the `bridge` network available to an outside network.
```

Comment:
Interesting thought while reading; do we want to be able to publish a port only on a specific network in future? @mavenugo ?